src/AmazonCloudDriveApi/AmazonCloudDriveApi.Shared/CloudDrive/CloudDriveApi.cs

```diff
@@ -188,37 +190,50 @@ public string GetDownloadUrl(string id)
 			return node.TempLink;
 		}
 
-		public async Task<CloudChangesResult> GetChanges(CloudChangesRequest parameters = null)
-		{
-			const string path = "changes";
-			try{
-				var url = CreateMetaUrl (path);
-				var postData = parameters == null || parameters.IsEmpty () ? "" : this.SerializeObject (parameters);
-				var message = await PostMessage (url, new StringContent (postData, Encoding.UTF8, "text/json"));
-				var data = await message.Content.ReadAsStringAsync ();
-				bool hasMore = false;
-
-				//Amazon sends horribly ugly/bad json format. It's actually 2 sets of json...
-				//To properly deseralize the data, you need to remove the bad data.
-				if (data.EndsWith ("{\"end\":true}")) {
-					data = data.Replace ("{\"end\":true}", "");
-					hasMore = false;
-				} else if (data.EndsWith ("{\"end\":false}")) {
-					data = data.Replace ("{\"end\":false}", "");
-					hasMore = true;
-				}
-				var result = Deserialize<CloudChangesResult> (data);
-				return result;
-			}
-			catch(Exception ex) {
-				return new CloudChangesResult {
-					Error = ex.ToString(),
-					ErrorDescription = ex.Message.ToString(),
-				};
-			}
-
-		}
-		#endregion //Nodes
+	    public async Task<bool> GetChanges(Func<CloudChangesResponse, bool> callback, CloudChangesRequest parameters = null)
+	    {
+	        const string path = "changes";
+
+	        var url = CreateMetaUrl(path);
+	        var postData = parameters == null || parameters.IsEmpty() ? "" : this.SerializeObject(parameters);
+	        var message = await PostMessage(url, new StringContent(postData, Encoding.UTF8, "text/json"));
+
+	        var contentBytes = await message.Content.ReadAsByteArrayAsync();
```

Comment:
I mentioned that in my original commit. You need to have a buffer implemented, and typically the buffer is based on the platform and the memory available. Because the get changes request can be in the 50-100MB, I have seen them this large. I did this for expediency to show how the code should be read.